### DISCUSSION TITLE
Problem with Xmodem on RS-232 port 1

### DISCUSSION THREAD

**Discussion by mmaseda12** [thehighnibble/thehighnibble.github.io#164] [IMSAI8080esp replica]
Trying to use Xmodem with IMSAI8080esp.  Using tera-term connected to RS-232 port 1 I get the CP/M prompt fine but when I run Xmodem and try a transfer it just hangs.  If I connect to RS-232 port 2 and run porta.sub and redirect the console to UC1 I get the CP/M prompt and Xmodem works fine.  I have both RS-232 ports are configure the same.  Any ideas?

Manuel

**Comment by mmaseda12** [comment]
Looking further I see there are some SIO parameters in the Xmodem.cfg file.  I’m sure that’s my problem.  I’ll report back with my findings.

Manuel

**Comment by udo-munk** [comment]
The first SIO used for the console is configured to strip parity and drop 0 bytes, because of some software sends text with parity set. xmodem is a 8 bit protocol, configure the first SIO same as the second, then it will work.

**Reply by mmaseda12** [reply]
Do I make this configuration change in Xmodem.cfg?  I’ve already made the changes in the esp config files.

Manuel

**Comment by udo-munk** [comment]
In the ESP config files. A SIO used for datatransfers must be 8bit clean, most protocols require this.

**Reply by mmaseda12** [reply]
I previously made these changes but it didn’t seem to work.  Is there anything else I need to change?

Manuel

Boot.cfg
UART0 = 115201 8N1

System.cfg
# SIO 1 Channel A, Ports 2/3 connected to terminal (default)
sio1a_upper_case 0
sio1a_strip_parity 0
sio1a_drop_nulls 0
sio1a_baud_rate 115200

**Comment by udo-munk** [comment]
This is correct for 8bit data passing the SIO without any modifications. The baud rate might be much to high for data transfers over a physical TTY port, I won't use much more than 9600. Probably your SIO2A is set for that and this is why it works on that one.

**Comment by mmaseda12** [comment]
Setting to 9600 still doesn’t works.  Since it’s working fine on the second port I I’ll just continue to use that.  Thanks for your help.

Manuel

**Comment by udo-munk** [comment]
Running out of ideas, sorry. This should work with both TTY ports, they are not different inside the ESP. Cannot test this my self, I have no RS-232 equipment anymore, doing everything with USB nowadays.

**Comment by mmaseda12** [comment]
Udo,
I tested this with Z80pack on a Pi5.  I used a serial console on the first serial port and configured it the same as I did on the ESP.  It worked fine.  Must be an issue with ESP.  I’ve asked Dave to look into it but said I was in no rush since I could used the second serial port.  Thanks again for your help.

Manuel
